### PR TITLE
[new content] Build-a-workflow-on-a-GitHub-hosted-runner

### PR DESCRIPTION
**Pull request by Fernando-Iscar**
Hi guys,

As discussed in our previous meeting, I've been working on this BB regarding a GitHub Actions workflow that automates code formatting and testing in your repository:

## Outline
- Set up GitHub Actions workflow.
- Implement code formatting and testing.

## Key Components
1. **Building a GitHub Actions Workflow**: Instructions for initializing a workflow.
2. **Code Formatting with Super Linter**: Automated code formatting checks.
3. **Code Testing with pytest**: Automated testing for code functionality.
4. **Workflow Integration and Tracking**: Adding the workflow to a repository and monitoring its execution.

---

Kindly have a look whenever you have time,

Waiting for your feedback,
Thank you!

@DiSanchz 
@shrabasteebanerjee 
@narendramukherjee

### CODE REVIEW

**Comment by DiSanchz** [comment]
I've just reviewed the Building Block and it looks great! Just have a few minor comments

File: content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
+title: "Build a workflow on a GitHub hosted runner"
```
Comment (DiSanchz) [inline content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md:2]: I think that the main takeaway of the BB for users is the implementation of the Testing/Formatting workflow rather than whether it runs on X or Y type of runner and the title would benefit from focusing on this. It could look something like:

"Build GitHub Actions workflow"
"Build a GitHub Actions workflow for code formatting and testing"
"Implement a GitHub Actions workflow for code formatting and testing"

Where "GitHub Actions" is given a central role in the title against references to the kind of runner.

File: content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+authorlink: "https://www.linkedin.com/in/fernando-iscar/"
+draft: false
+date: 2023-10-14T22:01:14+05:30 
+aliases: 
+  - /github_actions/workflow
+---
+
+## Overview
+
+This building block delves deeper into GitHub Actions by providing a hands-on example. We’ll craft a workflow specifically for code formatting and testing. After that, you'll discover how to integrate and track it into your own repository.
```
Comment (DiSanchz) [inline content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md:16]: This may not be strictly necessary but I think that adding a reference to Python code explicitly here may be interesting

"We’ll craft a workflow specifically for Python code formatting and testing. After that, you'll discover how to integrate and track it into your own repository."

File: content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+To illustrate a useful workflow example, our goal would be to integrate code formatting (using Super Linter) and code testing.
+
+In case you're interested, the example was retrieved from the following GitHub repository:
+
+{{% cta-primary-center "Click to go to the repository example" "https://github.com/snpe-reputation-systems/snpe-reputation-systems/blob/master/.github/workflows/formatting-testing.yml" %}}
+
+### Code Formatting with Super Linter
+
+[Super Linter](https://github.com/marketplace/actions/super-linter) is a simple combination of various linters to help validate source code. It works automatically whenever your workflow is activated by an event, like pushing new code.
```
Comment (DiSanchz) [inline content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md:40]: Again, this is good, but for further clarity I would rephrase to "The Super Linter action" at the beginning of the sentence for further clarity.

File: content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        VALIDATE_PYTHON_BLACK: true  # Enables Python formatting with Black
+        VALIDATE_PYTHON_ISORT: true  # Enables Python import sorting with isort
+
+```
+{{% /codeblock %}}
+
+This portion first checks out the code from the repository using `actions/checkout@v3`. After that, it uses the Super Linter action to validate the code. The environment variables under env specify the configuration for Super Linter.
+
+### Code Testing with pytest
+
```
[the remark this answers is not in the document]
Comment (DiSanchz) [reply content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md:84]: Sounds good to me! Perhaps we could add a tip box to introduce in one or two sentences the importance and usefulness of testing alongside the proposed link.

**Comment by Fernando-Iscar** [comment]
Thanks guys. All your suggestions made sense to me, so I've implemented them on my last commit.

If you don't have further ideas, I think we could merge it.

@narendramukherjee
@shrabasteebanerjee 
@DiSanchz
